pkg/proxy/handlers/spacelister_get.go

```diff
@@ -98,6 +142,43 @@ func GetUserWorkspace(ctx echo.Context, spaceLister *SpaceLister, workspaceName
 	), nil
 }
 
+// getUserSignupAndSpace returns the space and the usersignup for a given request
+func getUserSignupAndSpace(ctx echo.Context, spaceLister *SpaceLister, workspaceName string) (*signup.Signup, *toolchainv1alpha1.Space, error) {
+	userSignup, err := spaceLister.GetProvisionedUserSignup(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	space, err := spaceLister.GetInformerServiceFunc().GetSpace(workspaceName)
+	if err != nil {
+		ctx.Logger().Error(errs.Wrap(err, "unable to get space"))
+		return userSignup, nil, nil
```

Comment:
Why returning `nil` instead of `err`?

Reply:
I've left the functionality unchanged https://github.com/codeready-toolchain/registration-service/blob/81de976a7e62f89d73a44af3ffc61ce8188ab86e/pkg/proxy/handlers/spacelister_get.go#L52-L56

I see we return 404 when when usersignup and space are not found.

Reply:
Then let's add a comment to the function description stating that if no space found then a "nil" returned instead of an error.